### ISSUE TITLE
Large CPU usage of celery workers on edge version

### ISSUE THREAD

**Issue by mleneveut** [WeblateOrg/docker#443]
**Describe the bug**
I migrated from 3.6 to edge (to have the fix on redis SSL false and TLS verify false) and for 2 hours  now, the CPU of my host is at 100%, with celery workers taking 70% of it. The weblate website is working, but obviously, very slow.

We had only two very small repo in 3.6, so the migration should not be a problem.

**To Reproduce**
1. Have a 3.6 weblate with 2 small repositories
2. Migrate to weblate:edge

We are using Kubernetes, host is 2vCPU and 7Go memory. 

**Screenshots**
![cpu_celery](https://user-images.githubusercontent.com/3854415/68930561-f4152b80-078e-11ea-8ce9-ac1c6db1f7d7.png)

![host_cpu](https://user-images.githubusercontent.com/3854415/68930613-0db67300-078f-11ea-9bcb-b9a3ccb77a9f.png)

(yellow is iowait, purple is cpu system, blue is cpu user)

**Logs**

```
celery-main stderr | [2019-11-15 10:41:09,950: ERROR/MainProcess] Process 'ForkPoolWorker-92' pid:631 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:09,960: ERROR/MainProcess] Process 'ForkPoolWorker-91' pid:628 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:12,351: ERROR/MainProcess] Timed out waiting for UP message from <ForkProcess(ForkPoolWorker-90, started daemon)>
celery-main stderr | [2019-11-15 10:41:12,407: WARNING/MainProcess] process with pid=624 already exited
celery-main stderr | [2019-11-15 10:41:12,409: ERROR/MainProcess] Process 'ForkPoolWorker-94' pid:633 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:12,424: ERROR/MainProcess] Process 'ForkPoolWorker-93' pid:632 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:12,435: ERROR/MainProcess] Process 'ForkPoolWorker-90' pid:626 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:14,209: WARNING/MainProcess] process with pid=631 already exited
celery-main stderr | [2019-11-15 10:41:14,211: WARNING/MainProcess] process with pid=628 already exited
celery-main stderr | [2019-11-15 10:41:14,216: ERROR/MainProcess] Process 'ForkPoolWorker-96' pid:635 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:14,235: ERROR/MainProcess] Process 'ForkPoolWorker-95' pid:634 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:15,425: WARNING/MainProcess] process with pid=633 already exited
celery-main stderr | [2019-11-15 10:41:15,429: ERROR/MainProcess] Process 'ForkPoolWorker-99' pid:642 exited with 'signal 9 (SIGKILL)'
[2019-11-15 10:41:15,439: ERROR/MainProcess] Process 'ForkPoolWorker-98' pid:637 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:16,635: WARNING/MainProcess] process with pid=632 already exited
celery-main stderr | [2019-11-15 10:41:16,646: WARNING/MainProcess] process with pid=642 already exited
celery-main stderr | [2019-11-15 10:41:16,646: ERROR/MainProcess] Process 'ForkPoolWorker-101' pid:644 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:16,657: ERROR/MainProcess] Process 'ForkPoolWorker-100' pid:643 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:17,845: WARNING/MainProcess] process with pid=637 already exited
celery-main stderr | [2019-11-15 10:41:17,849: ERROR/MainProcess] Process 'ForkPoolWorker-103' pid:647 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:17,904: ERROR/MainProcess] Process 'ForkPoolWorker-102' pid:646 exited with 'signal 9 (SIGKILL)'
celery-memory stderr | [2019-11-15 10:41:32,173: INFO/MainProcess] missed heartbeat from notify@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:32,222: INFO/MainProcess] missed heartbeat from backup@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:32,222: INFO/MainProcess] missed heartbeat from search@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:32,222: INFO/MainProcess] missed heartbeat from main@weblate-c77dc9bcb-bhxvs
celery-search stderr | [2019-11-15 10:41:32,175: INFO/MainProcess] missed heartbeat from notify@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:32,222: INFO/MainProcess] missed heartbeat from memory@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:32,222: INFO/MainProcess] missed heartbeat from backup@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:32,223: INFO/MainProcess] missed heartbeat from main@weblate-c77dc9bcb-bhxvs
celery-backup stderr | [2019-11-15 10:41:32,174: ERROR/MainProcess] Process 'ForkPoolWorker-15' pid:581 exited with 'signal 9 (SIGKILL)'
celery-notify stderr | [2019-11-15 10:41:32,173: INFO/MainProcess] missed heartbeat from memory@weblate-c77dc9bcb-bhxvs
celery-notify stderr | [2019-11-15 10:41:32,229: INFO/MainProcess] missed heartbeat from backup@weblate-c77dc9bcb-bhxvs
celery-notify stderr | [2019-11-15 10:41:32,230: INFO/MainProcess] missed heartbeat from search@weblate-c77dc9bcb-bhxvs
celery-notify stderr | [2019-11-15 10:41:32,232: INFO/MainProcess] missed heartbeat from main@weblate-c77dc9bcb-bhxvs
celery-main stderr | [2019-11-15 10:41:32,232: ERROR/MainProcess] Process 'ForkPoolWorker-105' pid:649 exited with 'signal 9 (SIGKILL)'
celery-main stderr | [2019-11-15 10:41:32,243: ERROR/MainProcess] Process 'ForkPoolWorker-104' pid:648 exited with 'signal 9 (SIGKILL)'
celery-backup stderr | [2019-11-15 10:41:33,873: INFO/MainProcess] missed heartbeat from notify@weblate-c77dc9bcb-bhxvs
celery-backup stderr | [2019-11-15 10:41:33,874: INFO/MainProcess] missed heartbeat from memory@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:33,874: INFO/MainProcess] missed heartbeat from search@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:33,874: INFO/MainProcess] missed heartbeat from main@weblate-c77dc9bcb-bhxvs
celery-search stderr | [2019-11-15 10:41:59,521: INFO/MainProcess] missed heartbeat from main@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:59,521: INFO/MainProcess] missed heartbeat from memory@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:59,522: INFO/MainProcess] missed heartbeat from notify@weblate-c77dc9bcb-bhxvs
[2019-11-15 10:41:59,522: INFO/MainProcess] missed heartbeat from backup@weblate-c77dc9bcb-bhxvs
celery-notify stderr | [2019-11-15 10:41:56,176: INFO/MainProcess] missed heartbeat from main@weblate-c77dc9bcb-bhxvs
```

**Server configuration and status**

```
weblate list_versions
can't do it, doesn't respond due to 100% cpu

 weblate check --deploy
can't do it, doesn't respond due to 100% cpu
```

**Comment by mleneveut** [comment]
We are on Azure Kubernetes (AKS). Is there a ulimit or something to configure ? A health check ?

The pod yaml : 

```
spec:
  containers:
  - env:
    - name: WEBLATE_LOGLEVEL
      value: INFO
    - name: WEBLATE_SITE_TITLE
      value: Dekra translations
    - name: WEBLATE_ADMIN_NAME
      value: Weblate Admin
    - name: WEBLATE_ADMIN_EMAIL
      value: xxx@xxx.com
    - name: WEBLATE_ADMIN_PASSWORD
      value: xxx
    - name: WEBLATE_SERVER_EMAIL
      value: xxx@xxx.com
    - name: WEBLATE_DEFAULT_FROM_EMAIL
      value: weblate@xxx.com
    - name: WEBLATE_ALLOWED_HOSTS
      value: '*'
    - name: WEBLATE_REGISTRATION_OPEN
      value: "0"
    - name: SECURE_SSL_REDIRECT
      value: "true"
    - name: SECURE_HSTS_SECONDS
      value: "1209600"
    - name: SESSION_COOKIE_SECURE
      value: "true"
    - name: POSTGRES_HOST
      value: xxx.postgres.database.azure.com
    - name: POSTGRES_USER
      value: xxx@xxx.postgres.database.azure.com
    - name: POSTGRES_DATABASE
      value: weblate
    - name: POSTGRES_PORT
      value: "5432"
    - name: POSTGRES_PASSWORD
      valueFrom:
        secretKeyRef:
          key: password
          name: keycloak-db
    - name: REDIS_HOST
      value: xxx.redis.cache.windows.net
    - name: REDIS_PORT
      value: "6379"
    - name: REDIS_PASSWORD
      value: xxx
    - name: REDIS_TLS
      value: "false"
    - name: REDIS_VERIFY_SSL
      value: "false"
    - name: WEBLATE_EMAIL_HOST
    - name: WEBLATE_EMAIL_USER
    - name: WEBLATE_EMAIL_PASSWORD
    - name: KUBERNETES_PORT_443_TCP_ADDR
      value: xxx.hcp.northeurope.azmk8s.io
    - name: KUBERNETES_PORT
      value: tcp://xxx.hcp.northeurope.azmk8s.io:443
    - name: KUBERNETES_PORT_443_TCP
      value: tcp://xxx.hcp.northeurope.azmk8s.io:443
    - name: KUBERNETES_SERVICE_HOST
      value: xxx.hcp.northeurope.azmk8s.io
    image: weblate/weblate:edge
    imagePullPolicy: IfNotPresent
    name: weblate
    ports:
    - containerPort: 8080
      name: http
      protocol: TCP
    resources:
      limits:
        cpu: "1"
        memory: 1Gi
      requests:
        cpu: 100m
        memory: 800Mi
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /app/data
      name: weblate-data
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-fx2fx
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: aks-nodepool1-xxx
  nodeSelector:
    gitlab: "true"
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: weblate-data
    persistentVolumeClaim:
      claimName: weblate-data
  - name: default-token-fx2fx
    secret:
      defaultMode: 420
      secretName: default-token-fx2fx
```

**Comment by mleneveut** [comment]
I tried to downgrade to weblate:3.9.1-1 and it seems to fix the celery problem.

I set to true and the Redis connection works.

But I have this error : 

```
celery-search stderr | [2019-11-15 14:00:11,144: INFO/MainProcess] Connected to redis://:**@sf4v2ci.redis.cache.windows.net:6379/1
celery-search stderr | [2019-11-15 14:00:11,307: INFO/MainProcess] mingle: searching for neighbors
celery-backup stderr | [2019-11-15 14:00:11,448: INFO/MainProcess] sync with search@weblate-57f4879499-wdz2w
celery-backup stderr | [2019-11-15 14:00:11,554: ERROR/MainProcess] Control command error: OperationalError("\nCannot route message for exchange 'reply.celery.pidbox': Table empty or key no longer exists.\nProbably the key ('_kombu.binding.reply.celery.pidbox') has been removed from the Redis database.\n")
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/kombu/connection.py", line 439, in _reraise_as_library_errors
    yield
  File "/usr/local/lib/python3.7/dist-packages/kombu/connection.py", line 518, in _ensured
    return fun(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/kombu/messaging.py", line 203, in _publish
    mandatory=mandatory, immediate=immediate,
  File "/usr/local/lib/python3.7/dist-packages/kombu/transport/virtual/base.py", line 605, in basic_publish
    message, exchange, routing_key, **kwargs
  File "/usr/local/lib/python3.7/dist-packages/kombu/transport/virtual/exchange.py", line 70, in deliver
    for queue in _lookup(exchange, routing_key):
  File "/usr/local/lib/python3.7/dist-packages/kombu/transport/redis.py", line 877, in _lookup
    exchange, redis_key))
kombu.exceptions.InconsistencyError:
Cannot route message for exchange 'reply.celery.pidbox': Table empty or key no longer exists.
Probably the key ('_kombu.binding.reply.celery.pidbox') has been removed from the Redis database.
```

I delete the weblate deployment, flushdb the Redis database 1, drop database weblate on Postgres, created a new database et redeployed the weblate 3.9.1-1 but I have the same error.

Any idea ? Do I have to delete the PersistentVolumeClaim too ? (hosting the /app/data directory)

**Comment by nijel** [comment]
It's https://github.com/celery/kombu/issues/1063, it has been addressed by downgrading kombu in edge in 0983644942d0effb837e8bb3d3c738cf5105b7fb.

Back to killing the processes - maybe it's hitting some limits?

**Comment by mleneveut** [comment]
downgraded to version 3.8-5 and it works, I will stick to it until further problems or need ...

Thanks for the help.
